### PR TITLE
Short circuit to avoid DateTimeOffset parsing

### PR DESCRIPTION
**Pull request by BrennanConroy**
@JunTaoLuo

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
:shipit:

File: src/Microsoft.AspNetCore.ResponseCaching/ResponseCacheMiddleware.cs [hunk 1]
```diff
@@ -355,7 +355,7 @@ internal static bool ConditionalRequestSatisfied(ResponseCacheContext context)
                     }
                 }
             }
-            else if ((cachedResponseHeaders.LastModified ?? cachedResponseHeaders.Date) <= context.TypedRequestHeaders.IfUnmodifiedSince)
+            else if (context.TypedRequestHeaders.IfUnmodifiedSince != null && (cachedResponseHeaders.LastModified ?? cachedResponseHeaders.Date) <= context.TypedRequestHeaders.IfUnmodifiedSince)
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.ResponseCaching/ResponseCacheMiddleware.cs]: Remember that ResponseHeaders properties have no caching, they parse every time you access them

Comment (Tratcher) [inline src/Microsoft.AspNetCore.ResponseCaching/ResponseCacheMiddleware.cs]: and RequestHeaders...

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.ResponseCaching/ResponseCacheMiddleware.cs]: More cached fields on the context then.

**Pull request merged**